### PR TITLE
refactoring provider cluster spec

### LINKED ISSUE #9341: Refactor External cluster API model
Right now the External cluster object has `Spec` and `Cloud` with the specific provider:

 https://github.com/kubermatic/kubermatic/blob/4432afa00f60c8ba20fb570daa32ade11357e2d8/pkg/api/v2/types.go#L389

Each of the provider cloud spec has `ClusterSpec` which should be moved to the `ExternalClusterSpec`

```
// ExternalClusterSpec defines the external cluster specification.
type ExternalClusterSpec struct {
	// Version desired version of the kubernetes master components
	Version ksemver.Semver `json:"version,omitempty"`

	GKEClusterSpec *GKEClusterSpec `json:"gkeClusterSpec,omitempty"`
	EKSClusterSpec *EKSClusterSpec `json:"eksClusterSpec,omitempty"`
	AKSClusterSpec *AKSClusterSpec `json:"aksClusterSpec,omitempty"`
}
```
Fix all occurrences and tests: https://github.com/kubermatic/kubermatic/tree/4432afa00f60c8ba20fb570daa32ade11357e2d8/pkg/handler/v2/external_cluster

### PR DESCRIPTION
**Pull request by ahmadhamzh**
**What does this PR do / Why do we need it**:
move the provider cluster spec object from provider cloud spec to the external cluster spec

"closes #9341"

```release-note
NONE
```

### CODE REVIEW

**Comment by imharshita** [comment]
@ahmadhamzh Please update swagger, you can do that by running ./hack/update-swagger.sh in your local repo

**Comment by zreigz** [comment]
after `./hack/update-swagger.sh` run `./hack/gen-api-client.sh`

**Comment by imharshita** [comment]
> after `./hack/update-swagger.sh` run `./hack/gen-api-client.sh`

Also you can verify that everything is updated using
./hack/verify-api-client.sh and ./hack/verify-swagger.sh

File: pkg/handler/v2/external_cluster/external_cluster.go [hunk 1]
```diff
@@ -256,7 +258,7 @@ func CreateEndpoint(
 				}
 			}
 
-			createdCluster, err := createOrImportAKSCluster(ctx, req.Body.Name, userInfoGetter, project, cloud, clusterProvider, privilegedClusterProvider)
+			createdCluster, err := createOrImportAKSCluster(ctx, req.Body.Name, userInfoGetter, project, spec.AKSClusterSpec, cloud, clusterProvider, privilegedClusterProvider)
```
Comment (imharshita) [inline pkg/handler/v2/external_cluster/external_cluster.go]: check for nil pointer value spec.AKSClusterSpec , same for eks, gke

Comment (imharshita) [reply pkg/handler/v2/external_cluster/external_cluster.go]: Alternatively , just pass the spec here as in createorImport method we are checking if its nil

**Pull request merged**